crypto/evp/evp_utils.c

```diff
@@ -17,31 +17,66 @@
 #include "internal/evp_int.h"    /* evp_locl.h needs it */
 #include "evp_locl.h"
 
-int evp_do_ciph_getparams(const EVP_CIPHER *ciph, OSSL_PARAM params[])
-{
-    if (ciph->prov == NULL)
-        return -2;
-    if (ciph->get_params == NULL)
-        return -1;
-    return ciph->get_params(params);
+
+/*
+ * EVP_CTRL_RET_UNSUPPORTED = -1 is the returned value from any ctrl function
+ * where the control command isn't supported, and an alternative code path
+ * may be chosen.
+ * Since these functions are used to implement ctrl functionality, we
+ * use the same value, and other callers will have to compensate.
+ */
+#define EVP_PARAM_CHECK(obj, func, errfunc)                                    \
```

Comment:
Because these are only local macros, I see reason not to prefix them with `EVP_`.
I won't insist, but the way they are named now, they look like globally available macros...

Reply:
removed